core/src/consensus/tendermint/worker.rs

```diff
@@ -958,7 +959,7 @@ impl Worker {
     }
 
     pub fn on_imported_proposal(&mut self, proposal: &Header) {
-        if proposal.number() < 1 {
+        if self.client().block_status(&BlockId::Hash(*proposal.parent_hash())) == BlockStatus::Unknown {
```

Comment:
What do you think about checking whether the header is a snapshot header or not? The existence of a header's parent does not mean the header is a snapshot header.

Reply:
I added some comments explaining this near the updated code.